packages/fxa-payments-server/src/routes/Product/index.tsx

```diff
@@ -205,131 +172,108 @@ export const Product = ({
-          <p>{createSubscriptionStatus.error.message}</p>
-        </DialogMessage>
-      )}
-
-      <PaymentForm
-        {...{
-          onPayment,
-          onPaymentError,
-          onChangeErrorDismiss,
-          inProgress,
-          validatorInitialState,
-          confirm: true,
-          plan: selectedPlan,
-          onMounted: createSubscriptionMounted,
-          onEngaged: createSubscriptionEngaged,
-        }}
-      />
-    </div>
+    <SubscriptionCreate
+      {...{
+        profile: profile.result,
+        accountActivated,
+        selectedPlan,
+        createSubscriptionAndRefresh,
+        createSubscriptionStatus,
+        resetCreateSubscription,
+        validatorInitialState,
+        createSubscriptionMounted,
+        createSubscriptionEngaged,
+      }}
+    />
   );
 };
 
-type ProfileProps = {
-  profile: Profile;
+type PlansByIdType = {
+  [plan_id: string]: { plan: Plan; metadata: ProductMetadata };
 };
 
-const ProfileBanner = ({
-  profile: { email, avatar, displayName },
-}: ProfileProps) => (
-  <div className="profile-banner">
-    <img className="avatar hoisted" src={avatar} alt={displayName || email} />
-    {displayName && (
-      <h2 data-testid="profile-display-name" className="displayName">
-        {displayName}
-      </h2>
-    )}
-    <h3 data-testid="profile-email" className="name email">
-      {email}
-    </h3>
-    {/* TODO: what does "switch account" do? need to re-login and redirect eventually back here?
-      <a href="">Switch account</a>
-    */}
-  </div>
-);
+const indexPlansById = (plans: PlansFetchState): PlansByIdType =>
+  (plans.result || []).reduce(
+    (acc, plan) => ({
+      ...acc,
+      [plan.plan_id]: { plan, metadata: metadataFromPlan(plan) },
+    }),
+    {}
+  );
+
+// If the customer has any subscription plan that matches a plan for the
+// selected product, then they are already subscribed.
+const customerIsSubscribedToProduct = (
+  customerSubscriptions: Array<CustomerSubscription> | null,
+  productPlans: Plan[]
+) =>
+  customerSubscriptions &&
+  customerSubscriptions.some(customerSubscription =>
+    productPlans.some(plan => plan.plan_id === customerSubscription.plan_id)
+  );
 
-const AccountActivatedBanner = ({
-  profile: { email, displayName },
-}: ProfileProps) => (
-  <div data-testid="account-activated" className="account-activated">
-    <h2>
-      Your account is activated,{' '}
-      {displayName ? (
-        <>
-          <span data-testid="activated-display-name" className="displayName">
-            {displayName}
-          </span>
-        </>
-      ) : (
-        <>
-          <span data-testid="activated-email" className="email">
-            {email}
-          </span>
-        </>
-      )}
-    </h2>
-  </div>
-);
+// If the customer has any subscribed plan that matches the productSet
+// for the selected plan, then that is the plan from which to upgrade.
+const findUpgradeFromPlan = (
```

Comment:
A lot of long variable names here... I'm hoping that helps make the upgrade offer logic more clear, though.